plumbing/transport/fetch_pack.go

```diff
@@ -128,6 +128,28 @@ func (r *UploadPackRequest) Have(h ...plumbing.Hash) {
 	r.Haves = append(r.Haves, h...)
 }
 
+func (r *UploadPackRequest) IsEmpty() bool {
+	if len(r.Haves) < len(r.Wants) {
+		return false
+	}
```

Comment:
We do not perform an early return here. The early return is when IsEmpty = false. So, lines 132-134 are just a short-circuit in IsEmpty that avoids comparing every reference (unless the same reference is repeated multiple times in Wants.